### YamlMime:ManagedReference
items:
- uid: System.Xaml.XamlInternalException
  id: XamlInternalException
  children:
  - System.Xaml.XamlInternalException.#ctor
  - System.Xaml.XamlInternalException.#ctor(System.String)
  - System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: XamlInternalException
  nameWithType: XamlInternalException
  fullName: System.Xaml.XamlInternalException
  type: Class
  summary: "Исключение, которое выдается в случае внутренних несоответствий, возникающих во время чтения и записи XAML."
  remarks: "Это исключение обычно вызывается только в том случае, когда некоторые аспекты средства чтения XAML или системы средство записи XAML была расширена в коде, и несоответствия или проблемы реализации возникла, переданные системного кода служб XAML .NET Framework. Исключения, которые может быть создано сторонних производителей, например, синтаксический анализ кода XAML и обнаружения ошибок в XAML-ФАЙЛЕ, не должны вызывать XamlInternalException и вместо этого должны вызывать другие исключения, например <xref:System.Xaml.XamlParseException>или <xref:System.Xaml.XamlObjectWriterException>.</xref:System.Xaml.XamlObjectWriterException> </xref:System.Xaml.XamlParseException>       Если вы пишете код, который расширяет возможности служб XAML .NET Framework, в частности, при написании кода для средства чтения XAML или реализацию модуля записи XAML, необходимо соблюдать это различие между XamlInternalException и другие исключения, связанные с XAML. Код должен выдавать XamlInternalException только в случаях, когда это нарушает некоторые принцип или реализации шаблона разработки, вызываемых код определяет дальнейшее расширение класса. Код должен вызывать другие исключения, связанные с XAML для исключения случаев, когда средства чтения или записи реализации XAML используется без расширения и используется для загрузки XAML сторонних разработчиков, стороннего контекста схемы XAML и т. д."
  syntax:
    content: 'public class XamlInternalException : System.Xaml.XamlException'
  inheritance:
  - System.Exception
  - System.Xaml.XamlException
  implements: []
  inheritedMembers:
  - System.Xaml.XamlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlException.LineNumber
  - System.Xaml.XamlException.LinePosition
  - System.Xaml.XamlException.Message
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor
  id: '#ctor'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> класс системным сообщением, содержащим описание ошибки."
  syntax:
    content: public XamlInternalException ();
    parameters: []
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> класса с указанным сообщением, содержащим описание ошибки."
  syntax:
    content: public XamlInternalException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Сообщение, описывающее исключение."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> сериализованными данными."
  syntax:
    content: protected XamlInternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Объект, содержащий сериализованные данные объекта."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Контекстные сведения об источнике или назначении."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> класса с указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение."
  syntax:
    content: public XamlInternalException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Сообщение, описывающее исключение."
    - id: innerException
      type: System.Exception
      description: "Исключение, которое стало причиной текущего исключения. Если `innerException` не <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, текущее исключение сгенерировано в <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> блок, обрабатывающем внутреннее исключение."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Xaml.XamlException
  isExternal: false
  name: System.Xaml.XamlException
- uid: System.Xaml.XamlInternalException.#ctor
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Xaml.XamlInternalException.#ctor*
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException
  nameWithType: XamlInternalException.XamlInternalException
